frontend/app/directives.js

```diff
@@ -13,30 +13,134 @@ angular.module('TicTacToe.directives', [])
                     return $scope.game.currentPlayer;
                 };
                 this.playerIsLocal = function() {
-                    return $scope.player == 'local';
+                    return gameService.playerType == 'local';
                 };
                 this.gameID = function() {
                     return $scope.gameID;
                 };
                 this.move = function(boardRowIndex, boardColumnIndex, slotRowIndex, slotColumnIndex) {
-                    tictactoe.move($scope.game, boardRowIndex, boardColumnIndex, slotRowIndex, slotColumnIndex);
+                    return tictactoe.move($scope.game, boardRowIndex, boardColumnIndex, slotRowIndex, slotColumnIndex);
                 }
             }
         }
     }])
-    .directive('singleBoard', ['$http', 'gameService', function($http, gameService) {
+    .directive('boardHighlight', ['tictactoe', function(tictactoe) {
+        function linker(scope, element, attrs) {
+            var $el = $(element);
+            scope.curHighlightRow = undefined;
+            scope.curHighlightCol = undefined;
+
+            var setHighlight = function(boardRow, boardCol, cellRow, cellCol) {
+                var pos = $el.position();
+
+                scope.curHighlightRow = boardRow;
+                scope.curHighlightCol = boardCol;
+                console.log(boardRow, boardCol, cellRow, cellCol);
+
+                return new Promise(function(resolve, reject) {
```

Comment:
good to know, so i'll want to use these when my promise mutates state? i'll change to `$q` to be safe.